Benchmarks/main.swift

```diff
@@ -27,59 +29,46 @@ func runBenchmark(
         switch variety {
         case .inferenceThroughput:
             benchSettings =
-                settings.withDefaults(benchmarkModel.inferenceDefaults())
-            bench = benchmarkModel.inferenceBenchmark(benchSettings)
+                settings.withDefaults(benchmarkModel.defaultInferenceSettings)
+            bench = benchmarkModel.makeInferenceBenchmark(benchSettings)
             varietyName = "inference"
         case .trainingTime:
             benchSettings =
-                settings.withDefaults(benchmarkModel.trainingDefaults())
-            bench = benchmarkModel.trainingBenchmark(benchSettings)
+                settings.withDefaults(benchmarkModel.defaultTrainingSettings)
+            bench = benchmarkModel.makeTrainingBenchmark(benchSettings)
             varietyName = "training"
         }
-        benchmark(
+        let results = measure(
             name: "\(name) (\(varietyName))",
             settings: benchSettings,
             variety: variety,
-            benchmark: bench,
-            callback: logResults)
+            benchmark: bench)
+        reporter.printResults(results)
     } else {
         printError("No registered inference benchmark with a name: \(name)")
         printError("Consider running `list` command to see all available benchmarks.")
     }
 }
 
-func flags(from settings: BenchmarkSettings) -> String {
-    var result = ""
-    if settings.batches != -1 {
-        result += "--batches \(settings.batches) "
-    }
-    if settings.batchSize != -1 {
-        result += "--batchSize \(settings.batchSize) "
-    }
-    if settings.iterations != -1 {
-        result += "--iterations \(settings.iterations) "
-    }
-    if settings.epochs != -1 {
-        result += "--epochs \(settings.epochs) "
-    }
-    return result
-}
-
 let main =
     Group { group in
         group.command(
             "measure-all",
+            Flag("json", description: "Output json instead of plain text."),
             description: "Run all benchmarks with default settings."
-        ) {
+        ) { useJson in
```

Comment:
Minor: please capitalize all occurrences of "JSON", for consistency with other Swift JSON APIs:
```suggestion
        ) { useJSON in
```

Reply:
Fixed in a follow-up commit.